sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -268,6 +268,27 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
 
   object CartesianProduct extends Strategy {
     def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
+      // Not like the equal-join, BroadcastNestedLoopJoin doesn't support condition
+      // for cartesian join, as in cartesian join, probably, the records satisfy the
+      // condition, but exists in another partition of the large table, so we may not able
+      // to eliminate the duplicates.
```

Comment:
Does this comment related to this case? Also, I still do not think these two rules should be in this Strategy because the name of this strategy is `CartesianProduct`, but these first two cases are not for `CartesianProduct`. Actually, can we combine `CartesianProduct` and `BroadcastNestedLoopJoin` strategies?